packages/fluxible-addons-react/connectToStores.js

```diff
@@ -41,12 +42,16 @@ function createComponent(Component, stores, getStateFromStores, customContextTyp
                 this.context.getStore(Store).removeListener('change', this._onStoreChange);
             }, this);
         },
-        componentWillReceiveProps: function componentWillReceiveProps(nextProps){
-            this.setState(this.getStateFromStores(nextProps));
+        componentDidUpdate: function componentDidUpdate(prevProps, prevState) {
+            if (this.state._key === prevState._key) {
```

Comment:
what happens if we do not check the key?

Reply:
Than we would end in a recursive loop of setState -> componentDidUpdate -> setState. From [React docs](https://reactjs.org/docs/react-component.html#componentdidupdate) 

> You may call setState() immediately in componentDidUpdate() but note that it must be wrapped in a condition like in the example above, or you’ll cause an infinite loop.